kin-recovery/README.md

```diff
@@ -0,0 +1,180 @@
+- `onCancel` is called when the user is going back from the backup and restore activty to your activity.
+- `onError()` is called if there was some error in the process of backup or restore.
+
+```java
+// For backup
+backupAndRestoreManager.registerBackupCallback(new BackupCallback() {
+    @Override
+    public void onSuccess() {
+        // here you can handle the success.
+    }
+    
+    @Override
+    public void onCancel() {
+        // here you can handle the cancellation.
+    }
+    
+    @Override
+    public void onFailure(BackupException throwable) {
+        // here you can handle the failure.
+    }
+});
+
+// For restore  
+backupAndRestoreManager.registerRestoreCallback(new RestoreCallback() {
+    @Override
+    public void onSuccess(KinAccount kinAccount) {
+        // here you can handle the success.
+    }
+    
+    @Override
+    public void onCancel() {
+        // here you can handle the cancellation.
+    }
+    
+    @Override
+    public void onFailure(BackupException throwable) {
+        // here you can handle the failure.
+    }
+});
+```
+##### NOTE:
+Be sure to unregister from the module when no longer needed.  
+When you want to unregister then please use release:
+###### Snippet: unregister from the module and release all its resources
+```java 
+backupAndRestoreManager.release();
+``` 
+We recommend to register/unregister the callbacks in a way that will “survive” activity restart or similar situations.  
+For example, you can register in Activity.onCreate and release in Activity.onDestroy methods.  
+in MVP, for example, you can do it in the presenter Creation and Destruction.
+
+#### Step 3 - Passing the Result
+
+Because the module internally using `startActivityForResult` then in order for the module to work properly you need to implement `onActivityResult` in your activity and in that method you need to call
+`backupAndRestore.onActivityResult(...);`
+for example:
+```java 
+@Override
+protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+    super.onActivityResult(requestCode, resultCode, data);
+    backupAndRestore.onActivityResult(requestCode, resultCode, data);
+}
+```
+
+#### Step 4 - Backup or Restore
+
+Finally you can now use backup or restore flow by simply calling:
+
+```java 
+// For backup
+backupAndRestoreManager.backup(kinClient, kinAccount.getPublicAddress());
+
+// For restore
+backupAndRestoreManager.restore(kinClient)
+```
+
+### Error Handling
+
+onError(BackupAndRestoreException e) can be called in the case of an error that has occured while in the process of trying to backup or restore.
+
+### Testing
+
+Both unit tests and instrumented tests are provided.
+For a full list of tests see
+
+- https://github.com/kinecosystem/kin-sdk-android/tree/master/kin-recovery/recovery/src/test
+- https://github.com/kinecosystem/kin-sdk-android/tree/master/kin-recovery/recovery/src/androidTest
+
+
+### Running Tests
```

Comment:
probably not relevant for this specific module?

Reply:
why?
We have test and they can see them or use them, no?

Reply:
yep, I meant it sounds more overall project specific, specifically the below run_integ script that runs everything